server/src/main/java/org/opensearch/index/translog/TranslogWriter.java

```diff
@@ -460,7 +495,9 @@ public TranslogReader closeIntoReader() throws IOException {
                             channel,
                             path,
                             header,
-                            (translogCheckedContainer != null) ? translogCheckedContainer.getChecksum() : null
+                            translogWithoutFooterChecksum,
+                            translogWithFooterChecksum,
+                            true
```

Comment:
Why is it, that we are always passing `hasFooter` to `true` here? There can be cases when using older header and therefore, no footer right?

Reply:
This particular instance of `TranslogReader` creation is from `TranslogWriter.closeIntoReader`. This method is called only by current writers when they are rolling generation and creating readers out of them. Now, once this PR is merged, all the writers will have headers and therefore, whenever they try to close into reader, they will indeed have header.

Note that older header will never go through this workflow.

Reply:
ok. From what i understand by your comment. This below particular else code block looks redundant?
```
else {
                            // If we reach here then it means we are using older header and therefore, no footer.
                            // So, both translogWithoutFooterChecksum and translogWithFooterChecksum are same.
                            translogWithoutFooterChecksum = (translogCheckedContainer != null)
                                ? translogCheckedContainer.getChecksum()
                                : null;
                            translogWithFooterChecksum = translogWithoutFooterChecksum;
                        }
```

Reply:
This is there for safety purpose. Imagine a scenario wherein we have an open translog partially written when the `OpenSearch` process is upgraded, especially since it is under customer control. Now, when we will finally close this writer, then we will omit the footer in this case.

Now, for remote translog, we do not have a problem as the expectation is that any data which was in memory and not synced with remote will be lost. So the new footer will start anew.
But for local translog, the older partial translog will be present on disk which is of older version. So we still need to close it but not write the footer.